modules/fedimint-wallet-client/src/lib.rs

```diff
@@ -634,6 +652,50 @@ impl WalletClientModule {
         Ok(self.module_api.module_consensus_version().await? >= ModuleConsensusVersion::new(2, 2))
     }
 
+    /// Allocates a deposit address controlled by the federation, guaranteeing
+    /// safe handling of all deposits, including on-chain transactions exceeding
+    /// `ALEPH_BFT_UNIT_BYTE_LIMIT`.
+    ///
+    /// Returns an error if the client has never been online to verify the
+    /// federation's wallet module consensus version supports processing all
+    /// deposits.
+    pub async fn safe_allocate_deposit_address<M>(
+        &self,
+        extra_meta: M,
+    ) -> anyhow::Result<(OperationId, Address, TweakIdx)>
+    where
+        M: Serialize + MaybeSend + MaybeSync,
+    {
+        let mut dbtx = self.db.begin_transaction().await;
+
+        let already_verified_supports_safe_deposit =
+            dbtx.get_value(&SupportsSafeDepositKey).await.is_some();
+
+        let supports_safe_deposit = already_verified_supports_safe_deposit || {
+            match self.module_api.module_consensus_version().await {
+                Ok(module_consensus_version) => {
+                    let supported_version =
+                        SAFE_DEPOSIT_MODULE_CONSENSUS_VERSION <= module_consensus_version;
+
+                    if supported_version {
+                        dbtx.insert_new_entry(&SupportsSafeDepositKey, &()).await;
+                        dbtx.commit_tx().await;
+                    }
+
+                    supported_version
+                }
+                Err(_) => false,
+            }
+        };
```

Comment:
Can we make this a separate, pub fn such that integrators can use it to not show on-chain at all without attempting to generate an address?